### PR TITLE
Add OAuth

### PR DESCRIPTION
**Pull request by ahmedspiir**
We use Google Auth with OAuth which can be found here in [humio docs](https://docs.humio.com/cluster-management/security/oauth/).
And we were not able to set the values because it wasn't supported in the chart, so i have included OAuth type to humio-core.
Also passing secrets from values.yaml like `GOOGLE_OAUTH_CLIENT_SECRET` or anything else.
This block 
```yaml
{{- if .Values.env }}
{{ toYaml .Values.env | indent 12 }}
{{- end }}
```
Make it possible to reference secrets or pass environment variables so we will not be only tight to your vars.

### CODE REVIEW

File: charts/humio-core/templates/statefulset.yaml [hunk 1]
```diff
@@ -165,13 +165,15 @@ spec:
             - name: AUTO_CREATE_USER_ON_SUCCESSFUL_LOGIN
               value: {{ .Values.samlConfig.autoCreateUserOnSuccessfulLogin | quote }}
             {{- end }}
-            {{- if .Values.github.enabled }}
```
Comment (jswoods) [inline charts/humio-core/templates/statefulset.yaml]: We want to keep this. Perhaps this got removed inadvertently and requires a rebase from master?

**Comment by jswoods** [comment]
I added an `oauth` branch here if you want to take the 2nd route.

**Comment by ahmedspiir** [comment]
Hi @jswoods 
Thanks for the help :).
Regarding github value my intention was rather than always setting it from here, the chart user can set it an env variable and with this, you don't have to think about it. But i've restored it anyway.

**Comment by ahmedspiir** [comment]
I ran the script and uploaded the results, please check it

**Comment by ahmedspiir** [comment]
Hi @jswoods 
I've included a small adjustment to the cache setup to solve a problem that we've faced.
In Azure we don't have NVME setup for cache out of the box and we have to do some manual setup and run some scripts before starting humio.
This includes creating the PVs and creating the storage class `local-storage`, so in our case we don't want the chart to create the class for us.
I've introduced thins new option to the chart to make it more flexible and this fix solved our problem.

**Comment by jswoods** [comment]
@ahmedspiir very nice. That makes sense for local volume provisioner. I'm wondering if you mind adjusting two things, first make the `createStorageClass` default to true, so we don't break existing functionality. And the other thing is instead of `createStorageClass` could we call it `enableLocalVolumeProvisioner` or similar, since that's what it's doing (toggling the local volume provisioner on/off), again defaulting to true. What do you think?

**Comment by ahmedspiir** [comment]
@jswoods Thanks for your comment, makes total sense.
Can you check my changes :) ?

**Comment by jswoods** [comment]
@ahmedspiir looks great! Thank you for your contribution! Merging.

**Pull request merged**